### DISCUSSION TITLE
MySql upsert with `VALUES()` columns?

### DISCUSSION THREAD

**Discussion by GorgWorgington** [diesel-rs/diesel#3634] [Q&A] [answered]
Is it possible to update the conflicting rows with the values of the row that was attempted to be inserted on MySql?

Like this:
```sql
INSERT INTO t1 (a,b,c) VALUES (1,2,3),(4,5,6)
  ON DUPLICATE KEY UPDATE c=VALUES(a)+VALUES(b);
```
as shown here: https://dev.mysql.com/doc/refman/5.7/en/insert-on-duplicate.html

I see there's `diesel::upsert::excluded()` which seems like it would do what I want if I were using PostgreSQL.

**Comment by weiznich** [comment, marked as answer]
First of all: upsert support for mysql is a really new feature in diesel. It's not even released yet, so please expect that it might contain bugs or incomplete features.

That's currently not supported for the following reasons:

* There is no support for `excluded`/`VALUES` yet for the mysql backend
* Even if there where that support `excluded` does not support such an addition yet

That's both fixable and I would accept PR's for that.
